### PR TITLE
refactor: factor out some code to react-util and use it

### PR DESCRIPTION
**Pull request by hwbllmnn**
## Description

Makes use of react-util and factors out antd unspecific code to that library.

Switches the `npm ci` to use `--force` as the peer dependencies of client libraries are somewhat broken.

## Related issues or pull requests

Needs https://github.com/terrestris/react-util/pull/466

## Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the
  [BSD-2-Clause](https://github.com/terrestris/react-geo/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/react-geo/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [Code of Conduct](https://github.com/terrestris/react-geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

Please review @terrestris/devs

### CODE REVIEW

File: src/Button/CopyButton/CopyButton.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ const CopyButton: React.FC<CopyButtonProps> = ({
     if (digitizeLayer) {
       setLayers([digitizeLayer]);
     } else {
-      setLayers([DigitizeUtil.getDigitizeLayer(map)]);
+      setLayers([DigitizeUtil.getDigitizeLayer(map) as any]);
```
Comment (simonseyock) [inline src/Button/CopyButton/CopyButton.tsx:59]: why is this needed? Are the types of map not compatible? If so we should look if we can do something about it.

File: src/Map/FloatingMapLogo/FloatingMapLogo.tsx [hunk 2]
```diff
@@ -1,88 +1 @@
...
-        height={imageHeight}
-        style={imgStyle}
-      >
-      </img>
-    );
-  }
-}
-
-export default FloatingMapLogo;
+// placeholder
```
Comment (simonseyock) [inline src/Map/FloatingMapLogo/FloatingMapLogo.tsx:1]: can this get removed?

File: src/Hook/useMap.ts [hunk 3]
```diff
@@ -1,9 +1 @@
-import { useContext } from 'react';
-import OlMap from 'ol/Map';
-import MapContext from '../Context/MapContext/MapContext';
-
-export const useMap = (): OlMap|null => {
-  return useContext(MapContext);
-};
-
-export default useMap;
+// placeholder
```
Comment (simonseyock) [inline src/Hook/useMap.ts:1]: can this get removed?

File: src/Panel/SearchResultsPanel/SearchResultsPanel.tsx [hunk 4]
```diff
@@ -40,7 +38,12 @@ interface SearchResultsPanelProps extends Partial<CollapseProps> {
 
 const SearchResultsPanel = (props: SearchResultsPanelProps) => {
   const [highlightLayer, setHighlightLayer] = useState<OlLayerVector<OlSourceVector> | null>(null);
-  const map = useMap() as OlMap;
+  const map = useMap();
+
+  if (!map) {
+    return <></>;
```
Comment (simonseyock) [inline src/Panel/SearchResultsPanel/SearchResultsPanel.tsx:44]:
```suggestion
    return null;
```

**Comment by simonseyock** [comment]
very nice!

File: src/Map/FloatingMapLogo/FloatingMapLogo.tsx (hunk 2 above)
Comment (hwbllmnn) [reply src/Map/FloatingMapLogo/FloatingMapLogo.tsx:1]: Unfortunately not really. The styleguidist finds all examples that live near the matched `ts(x)` files. I can also specify the `.md` files directly, but then wildcards won't match, resulting in a much larger config.

File: src/Button/CopyButton/CopyButton.tsx (hunk 1 above)
Comment (hwbllmnn) [reply src/Button/CopyButton/CopyButton.tsx]: Fixed in latest amend

File: src/Hook/useMap.ts (hunk 3 above)
Comment (hwbllmnn) [reply src/Hook/useMap.ts:1]: See above

**Review by annarieger — APPROVED** [review]
